src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/dependency/GerritDependencyAction.java

```diff
@@ -0,0 +1,74 @@
+package com.sonyericsson.hudson.plugins.gerrit.trigger.dependency;
+
+import hudson.EnvVars;
+import hudson.model.AbstractBuild;
+import hudson.model.EnvironmentContributingAction;
+import hudson.model.InvisibleAction;
+import hudson.model.Run;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Adds Action that stores the data about dependency jobs.
+ */
+public class GerritDependencyAction extends InvisibleAction implements EnvironmentContributingAction {
+    @Nonnull
+    private final List<String> deps;
+
+    /**
+     * Saves the important information about the parent runs to be persistent.
+     * Stores string to keep the data even if original execution of parent job would be removed from history.
+     * @param runs List of runs it depend on
+     */
+    public GerritDependencyAction(List<Run> runs) {
+        deps = new ArrayList<String>(runs.size());
+        for (Run run : runs) {
+            deps.add(run.getParent().getFullName() + "#" + run.getNumber() + "#" + run.getResult());
+        }
+    }
+
+
+    /**
+     * The same as {@code buildEnvVars}, but for Run. Part of Core API since Jenkins 2.76.
+     * See https://issues.jenkins-ci.org/browse/JENKINS-29537 for more details.
+     *  @param run The calling build.
+     *  @param env Environment variables should be added to this map.
+     */
+    @SuppressWarnings("unused")
+    public void buildEnvironment(@Nonnull Run<?, ?> run, @Nonnull EnvVars env) {
+        fillEnv(env);
+    }
+
+    @Override
+    public void buildEnvVars(AbstractBuild<?, ?> build, EnvVars env) {
+        fillEnv(env);
+    }
+
+    /**
+     * Fills the environment variables based on build dependencies.
+     * @param env Environment variables should be added to this map.
+     */
+    private void fillEnv(EnvVars env) {
+        final StringBuilder depKeys = new StringBuilder();
+        for (String dependency : deps) {
+            String[] tokens = dependency.split("#");
+            String originalName = tokens[0];
+            String number = tokens[1];
+            String result = tokens[2];
+
+            String keyName = originalName.replaceAll("[^a-zA-Z0-9]+", "_");
+            String prefix = "DEP_" + keyName;
+
+            env.put(prefix + "_BUILD_NAME", originalName);
+            env.put(prefix + "_BUILD_NUMBER", number);
+            env.put(prefix + "_BUILD_RESULT", result);
+
+            depKeys.append(keyName);
+            depKeys.append(" ");
+            }
+
+        env.put("DEP_KEYS", depKeys.toString().trim());
```

Comment:
is `DEP_KEYS` really a good name?

Reply:
I did not want to put `GERRIT_` prefix in front to distinguish plugin "features" and data that comes from Gerrit. `DEP_KEYS` works fine for me, but we can rename it to something better. Any suggestions?

Reply:
Yea, that's a good distinction to have; that `GERRIT_*` is data about the event from Gerrit. I'll need to remember that :)
At the same time I think it need to be a bit more distinguishable from any other env var in the mix.

I'm not sure but maybe I just don't like the unnecessary abbreviation of dependency.

Perhaps somewhat mimicing what the [Parameterized Trigger Plugin](https://wiki.jenkins.io/display/JENKINS/Parameterized+Trigger+Plugin#ParameterizedTriggerPlugin-UsageasaBuildstep) provides would be OK? But perhaps not exactly.

`TRIGGER_DEPENDENCY_KEYS` ?

Reply:
And perhaps prefixing the others with `TRIGGER_` as well?

Reply:
I like this idea. Will switch to `TRIGGER_`.